dfc/tests/target_test.go

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+ *
+ */
+
+package dfc_test
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/NVIDIA/dfcpub/dfc"
+	"github.com/NVIDIA/dfcpub/pkg/client"
+)
+
+func TestPutObjectNoDaemonID(t *testing.T) {
+	const (
+		bucket = TestLocalBucketName
+		object = "someObject"
+	)
+	var sid string
+
+	smap, err := client.GetClusterMap(proxyurl)
+	checkFatal(err, t)
+
+	for sid = range smap.Tmap {
+		break
+	}
+
+	url := smap.Tmap[sid].DirectURL + "/" + dfc.Rversion + "/" + dfc.Robjects + "/" + bucket + "/" + object
+	err = client.HTTPRequest(http.MethodPut, url, nil)
+	if err == nil {
```

Comment:
Ideally, `client.HTTPRequest` would return some response object instead of just an `error` so that more things (ex. status code) can be checked. Without doing string parsing (which is not a reliable way to test things), I think this is more or less the best we can get.

Reply:
The similar thing I still have in backlog - make calls to DFC from client return normal HTTP error code. Otherwise it is hard to tell normal error, like object not found, from vague error, like internal server error